src/compiler/Uno.Compiler.Core/IL/Testing/TestSetupTransform.cs

```diff
@@ -88,11 +88,25 @@ public void CreateTestSetUpClass()
 
         Method CreateInvoker(Method method)
         {
+            if (!method.ReturnType.IsVoid || method.Parameters.Length > 0)
+            {
+                Log.Error(method.Source, null, "Tests must take zero parameters and return void");
+                return Function.Null;
+            }
+
+            // Promote non-public test methods to internal, to avoid accessibility problems
+            PromoteToInternal(method);
+
+            // Early-out on static methods, and classes with generated constructor not IDisposable
+            var ctor = method.DeclaringType.TryGetDefaultConstructor();
+            if (method.IsStatic || ctor.IsGenerated && !method.DeclaringType.IsImplementingInterface(Essentials.IDisposable))
+                return method;
+
             var invokeMethod = new Method(_source,
                                           _mainClass,
                                           null,
-                                          Modifiers.Generated | Modifiers.Public | Modifiers.Static,
-                                          InvokeName(method),
+                                          Modifiers.Generated | Modifiers.Private | Modifiers.Static,
+                                          method.FullName.Replace('.', '_').ToIdentifier(),
```

Comment:
Doesn`t this increase the risk name collisions? I think that's why we added the invoke-prefix. I'm not saying that's robust at all, but it seems slightly better than dropping it...

Reply:
I don't think I follow, but since the old code wasn't exactly iron-clad either... why not? :)